cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -107,17 +110,24 @@ Result<std::shared_ptr<ArrowType>> MakeArrowTimestamp(const LogicalType& logical
   }
 }
 
-Result<std::shared_ptr<ArrowType>> FromByteArray(const LogicalType& logical_type) {
+Result<std::shared_ptr<ArrowType>> FromByteArray(
+    const LogicalType& logical_type, const ArrowReaderProperties& reader_properties) {
   switch (logical_type.type()) {
     case LogicalType::Type::STRING:
       return ::arrow::utf8();
     case LogicalType::Type::DECIMAL:
       return MakeArrowDecimal(logical_type);
     case LogicalType::Type::NONE:
     case LogicalType::Type::ENUM:
-    case LogicalType::Type::JSON:
     case LogicalType::Type::BSON:
       return ::arrow::binary();
+    case LogicalType::Type::JSON:
+      if (reader_properties.get_arrow_extensions_enabled()) {
```

Comment:
Sorry that I'm late on the review. Is this property too board? Should we restrict it to json specifically?

Reply:
I'm not sure why would we want a more specific property. Could you say more?
Thinking about it - `json` extension doesn't have any extra requirements so we'll likely always have it enabled. Same for `bool8` and `uuid`. On the other hand `fixed_shape_tensor` and `opaque` depend on `rapidjson` so they would not always be available, but they are not Parquet logical types.